gateway/app/main/routes.py

```diff
@@ -332,3 +333,99 @@ def download_model():
         return Response(
             json.dumps(response_body), status=status_code, mimetype="application/json"
         )
+
+
+@main.route("/req_join", methods=["GET"])
+def fl_cycle_application_decision():
+    """
+        use the temporary req_join endpoint to mockup:
+        - reject if worker does not satisfy 'minimum_upload_speed' and/or 'minimum_download_speed'
+        - is a part of current or recent cycle according to 'do_not_reuse_workers_until_cycle'
+        - selects according to pool_selection
+        - is under max worker (with some padding to account for expected percent of workers so do not report successfully)
+    """
+
+    # parse query strings (for now), evetually this will be parsed from the request body
+    model_id = request.args.get("model_id")
+    up_speed = request.args.get("up_speed")
+    down_speed = request.args.get("down_speed")
+    worker_id = request.args.get("worker_id")
+    _cycle = processes.get_cycle(model_id)
+    _accept = False
+    """
+    MVP variable stubs:
+        we will stub these with hard coded numbers first, then make functions to dynaically query/update in subsquent PRs
+    """
+    # this will be replaced with a function that check for the same (model_id, version_#) tuple when the worker last participated
+    last_participation = 1
+    # how late is too late into the cycle time to give a worker "new work", if only 5 seconds left probably don't bother
+    MINIMUM_CYCLE_TIME_LEFT = 500
```

Comment:
Not a bad idea, but we'll want to more intelligently determine this in the future